### ISSUE TITLE
Positional parameters

### ISSUE THREAD

**Issue by IvanIvanoff** [appodeal/clickhouse_ecto#8]
I see that you removed the `?<number>` in https://github.com/appodeal/clickhouse_ecto/pull/7
My use case requires a lot of repeating parameters, so I forked the repo and implemented it this way https://github.com/santiment/clickhouse_ecto/pull/1 which I believe is the better way to handle it.

Are you planing to go forward without supporting positional parameters at all?

**Comment by scarfacedeb** [comment]
@IvanIvanoff could you provide a code example for ecto query with multiple params with the same `?1` placeholder? I couldn't think of such cases in our code, that's why I though it will be safe to remove positional arguments. (like in mysql adapter)

**Comment by IvanIvanoff** [comment]
@scarfacedeb For one of my use cases I'm writing plain sql and execute it with `ClickhouseRepo.query`.
Example is to fill gaps when bucketing into time intervals see this code:
```
query = """
    SELECT SUM(value), time
    FROM (
      SELECT
        toDateTime(intDiv(toUInt32(?4 + number * ?1), ?1) * ?1) as time,
        toFloat64(0) AS value
      FROM numbers(?2)
      UNION ALL
      SELECT toDateTime(intDiv(toUInt32(dt), ?1) * ?1) as time, sum(value) as value
      FROM transfers
      PREWHERE from IN (?3) AND NOT to IN (?3)
      AND dt >= toDateTime(?4)
      AND dt <= toDateTime(?5)
      GROUP BY time
      ORDER BY time
    )
    GROUP BY time
    ORDER BY time
    """
    args = [
      interval,
      span,
      wallets,
      from_datetime_unix,
      to_datetime_unix
    ]
```

**Comment by scarfacedeb** [comment]
On second thought, `clickhouse` itself doesn't support any placeholder params at all.

Therefore any params-related features will be implemented in the `clickhousex` only, without the db support. 

On the one hand, It gives certain flexibility to the implementation. On the other hand, it means extra support for that feature.

In contrast to `postgres` that supports ordered params out of the box, and therefore it makes sense to use them in `postgrex`.

I guess only the maintainer can tell if it's worth pursuing.

**Comment by IvanIvanoff** [comment]
@scarfacedeb I'm still not convinced where it should be handled. `clickhouse_ecto` translates the query from containing `?<num>` to plain query containing plain `?` and updates the argument list accordingly. In my opinion `clickhousex` should implement only syntax that ClickHouse understands.

Are you aware of any such situation in other adapters and how they handle it? 
Now when I think about it I'm also not sure if adding features not supported by ClickHouse brings more benefit or more harm.

**Comment by scarfacedeb** [comment]
@IvanIvanoff 
> Now when I think about it I'm also not sure if adding features not supported by ClickHouse brings more benefit or more harm.

yep, that's exactly what I was thinking in the message above :) 

> Are you aware of any such situation in other adapters and how they handle it?

As far as my understanding goes, 

postgrex supports positional params, because postgres itself supports them (see: https://github.com/elixir-ecto/postgrex/blob/43aa6b964043dd8726dba363245aebd3734bde47/lib/postgrex/messages.ex#L289)

`Ecto.Adapters.Mysql` only supports `?` params. (and it looks like `mariaex` sends those params directly: https://github.com/xerions/mariaex/blob/16d8abcfe479005651bd11050303382448664e6a/lib/mariaex/protocol.ex#L487)

**Comment by santaux** [comment]
@scarfacedeb 

> what do you think?
I think that you were right when said that clickhouse itself doesn't support any placeholder params at all. And also, as @IvanIvanoff said:
> Now when I think about it I'm also not sure if adding features not supported by ClickHouse brings more benefit or more harm.
:)
